src/operator/controllers/intents_reconcilers/internet_network_policy/internet_network_policy.go

```diff
@@ -365,3 +282,42 @@ func (r *InternetNetworkPolicyReconciler) buildPodLabelSelectorFromIntents(inten
 		},
 	}
 }
+
+// applyServiceEffectivePolicy - reconcile ingress netpols for a service. returns the list of policies' namespaced names
+func (r *InternetNetworkPolicyReconciler) applyServiceEffectivePolicy(ctx context.Context, ep effectivepolicy.ServiceEffectivePolicy) ([]types.NamespacedName, error) {
+	hasAnyInternetIntents := slices.ContainsFunc(ep.ClientIntent.GetCallsList(), func(intent otterizev1alpha3.Intent) bool {
+		return intent.Type == otterizev1alpha3.IntentTypeInternet
+	})
+	if !hasAnyInternetIntents || !ep.ClientIntent.DeletionTimestamp.IsZero() {
+		return nil, nil
```

Comment:
Return an empty slice instead of a nil slice
1. Clearer - shows it wasn't a mistake
2. Will not panic under any circumstance